components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/SequenceGenerator.java

```diff
@@ -0,0 +1,408 @@
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.wso2.carbon.apimgt.impl.utils.APIUtil.handleException;
+
+/**
+ * Class uses to generate api sequences for soap to rest conversion.
+ */
+public class SequenceGenerator {
+    private static final Logger log = LoggerFactory.getLogger(SequenceGenerator.class);
+
+    private static APIDefinition definitionFromSwagger20 = new APIDefinitionFromSwagger20();
+
+    /**
+     * generates api in/out sequences from the swagger to soap definitions
+     * <p>
+     * Note: this method is directly invoked from the jaggery layer
+     *
+     * @param apiDataStr           api object as a string
+     * @param soapOperationMapping soap operation mapping from the wsdl
+     * @throws APIManagementException throws if error occurred in registry access/ parsing definitions
+     */
+    public static void generateSequences(String apiDataStr, String soapOperationMapping) throws APIManagementException {
+        JSONParser parser = new JSONParser();
+        boolean isTenantFlowStarted = false;
+        try {
+            JSONObject apiData = (JSONObject) parser.parse(apiDataStr);
+            String provider = (String) apiData.get("provider");
+            String name = (String) apiData.get("name");
+            String version = (String) apiData.get("version");
+
+            if (provider != null) {
+                provider = APIUtil.replaceEmailDomain(provider);
+            }
+
+            provider = (provider != null ? provider.trim() : null);
+            name = (name != null ? name.trim() : null);
+            version = (version != null ? version.trim() : null);
+            APIIdentifier apiId = new APIIdentifier(provider, name, version);
+
+            JSONObject apiJSON;
+
+            String tenantDomain = MultitenantUtils.getTenantDomain(APIUtil.replaceEmailDomainBack(provider));
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            }
+            RegistryService registryService = ServiceReferenceHolder.getInstance().getRegistryService();
+            int tenantId;
+            UserRegistry registry;
+            try {
+                tenantId = ServiceReferenceHolder.getInstance().getRealmService().
+                        getTenantManager().getTenantId(tenantDomain);
+                registry = registryService.getGovernanceSystemRegistry(tenantId);
+
```

Comment:
We need to load the tenant registry if its tenant before getting the registry object

[1] - https://github.com/wso2/carbon-apimgt/blob/6.x/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java#L4401